### PR TITLE
Migrate the `BoxGrid` component to Chakra

### PR DESCRIPTION
**Pull request by soheil555**
## Description

Migrate the `BoxGrid` component to Chakra

## Related Issue
#8632

### CODE REVIEW

**Review by TylerAPfledderer — CHANGES_REQUESTED** [review]
Hey @soheil555! Nice work here. 🤘🏼 

For you and @pettinarip, I've provided some thoughts with using `SimpleGrid` to keep from flipping between grid and flexbox, along with tricks to simplify the code. Wondering from both of you if these suggested changes improve readability. 😄

**Comment by soheil555** [comment]
Hey @TylerAPfledderer 
Awesome suggestions!
Thank you so much.

File: src/components/BoxGrid.tsx [hunk 1]
```diff
@@ -111,66 +35,81 @@ const colors = [
...
 
   return (
-    <Grid>
+    <Box
+      display={{ base: "flex", lg: "grid" }}
+      gridTemplateColumns="repeat(4, 1fr)"
+      flexDirection="column"
+      my={16}
+      borderRadius="2px"
+    >
```
[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/BoxGrid.tsx:48]: This totally simplifies the component overall. Nice suggestion!

**Review by pettinarip — CHANGES_REQUESTED** [review]
Thanks @soheil555, the component is working perfectly for me 💯 

Would like to see the `SimpleGrid` implementation here if that is possible.

Thanks @TylerAPfledderer for the suggestions and review 💪🏼

**Review by pettinarip — APPROVED** [review]
🚢

**Pull request merged**